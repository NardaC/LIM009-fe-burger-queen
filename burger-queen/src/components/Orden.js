import React from "react";

const Orden = ({ setPedidos, setClientName, setTableNumber }) => (
  <div className="trans text-center">
    <button
      data-testid="new-order-btn"
      type="button"
      className="btn btn-primary mx-2 px-5 mb-3 mt-3 aos-init aos-animate"
      onClick={() => {
        setPedidos([]);
        setClientName("");
        setTableNumber("");
      }}
    >
      Nueva Orden
    </button>
    <button
      data-testid="order-list-btn"
      type="button"
      className="btn btn-primary mx-2 px-5 mb-3 mt-3 aos-init aos-animate"
    >
      Lista de ordenes
    </button>
  </div>
);
export default Orden;
