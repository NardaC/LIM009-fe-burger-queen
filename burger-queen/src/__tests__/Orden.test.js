import React from 'react';
import { render, fireEvent, getByAltText } from '@testing-library/react';
import '@testing-library/jest-dom';
import '@testing-library/react/cleanup-after-each'
import '@testing-library/jest-dom/extend-expect'
import Orden from '../components/Orden';

it('renders', () => {
    const { asFragment } = render( <Orden/>);
    expect(asFragment()).toMatchSnapshot();
})
it('inserts logo image in <img>', () => {
    const{ getByTestId, getByText} =render (<Orden />)
    expect(getByTestId('new-order-btn')).toHaveClass("btn btn-primary mx-2 px-5 mb-3 mt-3 aos-init aos-animate");
    expect(getByTestId('order-list-btn')).toHaveClass("btn btn-primary mx-2 px-5 mb-3 mt-3 aos-init aos-animate");
    expect(getByText('Nueva Orden')).toHaveAttribute('type', expect.stringContaining('button'));
})
