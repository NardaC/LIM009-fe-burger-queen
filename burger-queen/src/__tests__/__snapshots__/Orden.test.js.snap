// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="trans text-center"
  >
    <button
      class="btn btn-primary mx-2 px-5 mb-3 mt-3 aos-init aos-animate"
      data-testid="new-order-btn"
      type="button"
    >
      Nueva Orden
    </button>
    <button
      class="btn btn-primary mx-2 px-5 mb-3 mt-3 aos-init aos-animate"
      data-testid="order-list-btn"
      type="button"
    >
      Lista de ordenes
    </button>
  </div>
</DocumentFragment>
`;
